File: clients/algoliasearch-client-csharp/algoliasearch/Clients/ClientOptions.cs
using System;
using System.Collections.Generic;
using Algolia.Search.Models.Common;
using Algolia.Search.Transport;

namespace Algolia.Search.Clients;

/// <summary>
/// Optional overrides for the ingestion transporter created by <c>*WithTransformation</c> helpers.
/// Only fields explicitly set here replace the Ingestion API defaults (25 s timeouts, region-derived hosts, no compression).
/// Credentials and region are always taken from the parent <see cref="TransformationOptions"/>.
/// See https://www.algolia.com/doc/libraries/sdk/methods/ingestion
/// </summary>
public class ClientOptions
{
  /// <summary>Override the default ingestion hosts.</summary>
  public List<StatefulHost> CustomHosts { get; set; }

  /// <summary>Override the default connect timeout (25 s).</summary>
  public TimeSpan? ConnectTimeout { get; set; }

  /// <summary>Override the default read timeout (25 s).</summary>
  public TimeSpan? ReadTimeout { get; set; }

  /// <summary>Override the default write timeout (25 s).</summary>
  public TimeSpan? WriteTimeout { get; set; }

  /// <summary>Override the default compression (none).</summary>
  public CompressionType? Compression { get; set; }

  /// <summary>Override the default headers.</summary>
  public Dictionary<string, string> DefaultHeaders { get; set; }
}

File: clients/algoliasearch-client-swift/Sources/Core/Networking/ClientOptions.swift
//
//  ClientOptions.swift
//
//  Overridable client options shared across the SDK. Every field is optional; an unset field
//  keeps the calling client's default. Used as-is by `TransformationOptions` to override the
//  ingestion transporter defaults (25 s timeouts, region-derived hosts, no compression).
//
//  See https://www.algolia.com/doc/libraries/sdk/methods/ingestion

import Foundation

public struct ClientOptions {
    /// Override the read timeout.
    public var readTimeout: TimeInterval?

    /// Override the write timeout.
    public var writeTimeout: TimeInterval?

    /// Override the hosts.
    public var hosts: [RetryableHost]?

    /// Override the compression.
    public var compression: CompressionAlgorithm?

    /// Additional headers merged into every request.
    public var defaultHeaders: [String: String]?

    /// - parameter readTimeout: Override the read timeout.
    /// - parameter writeTimeout: Override the write timeout.
    /// - parameter hosts: Override the hosts.
    /// - parameter compression: Override the compression.
    /// - parameter defaultHeaders: Additional headers merged into every request.
    public init(
        readTimeout: TimeInterval? = nil,
        writeTimeout: TimeInterval? = nil,
        hosts: [RetryableHost]? = nil,
        compression: CompressionAlgorithm? = nil,
        defaultHeaders: [String: String]? = nil
    ) {
        self.readTimeout = readTimeout
        self.writeTimeout = writeTimeout
        self.hosts = hosts
        self.compression = compression
        self.defaultHeaders = defaultHeaders
    }
}
